config/config.go

```diff
@@ -359,6 +374,15 @@ func convertToTerragruntConfig(terragruntConfigFromFile *terragruntConfigFile, t
 		terragruntConfig.RemoteState = terragruntConfigFromFile.RemoteState
 	}
 
+	//Validate before and after hooks to make sure that they are properly configured
+	allHooks := append(terragruntConfigFromFile.Terraform.BeforeHooks, terragruntConfigFromFile.Terraform.AfterHooks...)
+
+	for _, curHook := range allHooks {
+		if curHook.Execute[0] == "" {
```

Comment:
Could `curHook.Execute` be an empty list? So that this throws an out of bounds error?

Reply:
Good point - Added a new test for this and fixed the underlying issue